internal/handlers/commonpath/path.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commonpath
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/iterator"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// FindValuesOpts sets options for FindValues.
+type FindValuesOpts struct {
+	// FindArrayIndex gets an element at the specified index of an array.
+	// Using path `v.0` and `v` is an array, it returns 0-th index element of the array.
+	// If `v` is not an array, FindArrayIndex has no impact.
+	FindArrayIndex bool
+	// SearchArray searches all document of an array to find documents that contains path key.
+	// Using path `v.foo` and `v` is an array, it returns all document which has key `foo`.
+	// If `v` is not an array, SearchArray has no impact.
+	SearchArray bool
```

Comment:
Should this one be called `SearchInArray` or `IterateThroughArray`? (But for this second option I actually liked the previous one with `Ignore` because we were really ignoring things in this case)

In this case, we don't aim to search for an array, we rather iterate through it.

And the doc is something like "If `SearchInArray` is set, the search will iterate through the whole array to find matches in any of its elements".

Reply:
Renamed this to `SearchInArray`. For ignoring other one, I'm not totally sure to have one option that enables array search and another option disables index search. :thinking:

Reply:
@rumyantseva, it went though a few iteration of renaming and re-writing. Could you resolve this comment?